Node.js/boilerplate.js

```diff
@@ -413,15 +430,99 @@ class MQBoilerPlate {
     return `${env.MQ_ENDPOINTS[i].HOST}(${env.MQ_ENDPOINTS[i].PORT})`;
   }
 
-  buildMQCNO() {
+  jwtCheck(MQDetails) {
+
+    if (!env.JWT_ISSUER) {
+
+      debug_info('JWT credentials not found, will not be using JWT to authenticate');
+      return false;
+
+    } else if (MQDetails.JWT_TOKEN_ENDPOINT === null || MQDetails.JWT_TOKEN_USERNAME === null || 
+        MQDetails.JWT_TOKEN_PWD === null || MQDetails.JWT_TOKEN_CLIENTID === null) {
+
+      debug_info('One or more JWT credentials missing, will not be using JWT to authenticate');
+      return false;
+    } 
+
+    debug_info('JWT credentials found, will be using JWT to authenticate');
+    return true;
+  }
+
+  async obtainToken() {
+    // Asynchronous function to handle http & https requests
+    let me = this;
+    let accessToken;
+
+    debug_info('Obtaining token from:', me.MQDetails.JWT_TOKEN_ENDPOINT);
+    
+    let formData = querystring.stringify({
+      username: me.MQDetails.JWT_TOKEN_USERNAME,
+      password: me.MQDetails.JWT_TOKEN_PWD,
+      client_id: me.MQDetails.JWT_TOKEN_CLIENTID,
+      grant_type: "password",
+    });
+
+    // Regex to extract the hostname:port out of the endpoint
+    // This is used to ensure the correct URL is used for the request
+    // and the JWT obtained has a valid 'iss' claim
+    let match = me.MQDetails.JWT_TOKEN_ENDPOINT.match(/^https?:\/\/([^/]+)/i);
+    let hostPort = match ? match[1] : null;
+    
+    // Creating server request
+    let options = {
+      method: 'POST',
+      url: me.MQDetails.JWT_TOKEN_ENDPOINT,
+      data: formData,
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        "Content-Length": formData.length,
+        "Host": hostPort,
+      }
+    }
+
+    // Checking for JWT authentication with JWKS
+    // If key repository is enabled, the https server (token issuer) can be trusted
+    // with this public certificate
+    if (me.MQDetails.JWT_KEY_REPOSITORY) {
+      options.httpsAgent = new https.Agent({ ca: fs.readFileSync(me.MQDetails.JWT_KEY_REPOSITORY)});
+    }
+      
+    // Sending request to token server
+    // Wait until response received
+    let response = await axios(options);
```

Comment:
Have you tested with an error - eg wrong host / port specified, to see what happens when the Axios promise is rejected?